### PR TITLE
Upgrade BAM collapsing workflow

### LINKED ISSUE #142: Upgrade BAM Collapsing Workflow
Upgrade CWL to the new version using the Flowchart attached and using the BAM Collapsing Section in the Flowchart.

Fgbio Workflow_edited

Note: Flowchart with contents in Teal color are to be taken from cwl-commandlinetools and the rest are from cwl_subworkflows.

- [ ] CWL
- [ ] Readme
- [ ] Example Input
- [ ] Update Gitbook

### PR DESCRIPTION
**Pull request by rhshah**
Resolves #142

### CODE REVIEW

**Comment by murphycj2** [comment]
is this ready for review? because i see the readme, example inputs, etc missing

File: bam_collapsing_marianas.cwl [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+      standard. It has multiple secondary file associated with it ending in
+      ".dict, .fai, .amb, .ann, .bwt, .pac, .index, .rbwt, .rpac, .rsa, .sa"
+    secondaryFiles:
+      - .fai
+      - ^.dict
+      - .amb
+      - .ann
+      - .bwt
+      - .pac
+      - .sa
```
Comment (ionox0) [inline bam_collapsing_marianas.cwl:22]: Similar to the new standard bam processing workflow, I'm not sure which of these is required, but if some of them are optional we could consider making them so (I think we should be able to use `type: File` as part of `secondaryFiles`)

Comment (ionox0) [reply bam_collapsing_marianas.cwl:22]: Just realized this is the Marianas workflow, so probably not necessary to change actually

File: bam_collapsing.cwl [hunk 2]
```diff
@@ -1,602 +1,817 @@
...
+    'sbg:y': 4909.0078125
+  - id: bwa_mem_t
+    type: int?
+    'sbg:x': 0
+    'sbg:y': 5015.6015625
+  - id: sort_order
+    type: string?
+    'sbg:x': 0
+    'sbg:y': 106.6640625
+  - id: bwa_mem_sample_id
```
Comment (ionox0) [inline bam_collapsing.cwl]: I guess this can be made as optional (I'll fix the commandlinetool soon)

Comment (ionox0) [reply bam_collapsing.cwl]: Or it can be called just `sample_id` if it is used in ARRG

File: bam_collapsing.cwl [hunk 3]
```diff
@@ -1,602 +1,817 @@
...
+    'sbg:y': 2780.28125
+  - id: fgbio_filter_consensus_read_output_file_name_simplex_duplex
+    type: string?
+    'sbg:x': 2182.387451171875
+    'sbg:y': 2246.609375
+  - id: fgbio_filter_consensus_read_output_file_name_simplex_aln_metrics
+    type: string?
+    'sbg:x': 2182.387451171875
+    'sbg:y': 2353.34375
+  - id: fgbio_postprocessing_output_file_name_simpex
```
Comment (ionox0) [inline bam_collapsing.cwl]: "simpex" --> "simplex"

File: bam_collapsing.cwl [hunk 4]
```diff
@@ -1,602 +1,817 @@
...
-      - gzip_Read1/output_file
-    type: File
-    'sbg:x': 1386.576171875
-    'sbg:y': 1509.640625
-  - id: gzip_read2
+      - gatk_sam_to_fastq_4_1_8_0/gatk_sam_to_fastq_second_end_fastq
+    type: File?
+    'sbg:x': 2182.387451171875
+    'sbg:y': 1499.46875
+  - id: gatk_sam_to_fastq_fastq
```
Comment (ionox0) [inline bam_collapsing.cwl:639]: Do we need to save the fastqs from GATK samtofastq? maybe we want to check that the UMIs are correctly formatted in the read header? or maybe we don't need to check them

Comment (ionox0) [reply bam_collapsing.cwl:639]: Sorry, just remembered these would be the collapsed fastqs, which probably would make sense to save

**Pull request merged**